### PR TITLE
fix file not found error when deploy contract twice

### PR DESCRIPTION
**Pull request by tyrelltle**
@judezhu
use npm --prefix arg to specify package path when combine file with orable.
improved error handling to prevent server crash

### CODE REVIEW

File: utils/build.js [hunk 1]
```diff
@@ -90,15 +90,19 @@ exports.mergeCrowdsaleContract = async function (tokenId) {
...
-        shell.echo('Error: merge file failed');
-        shell.exit(1);
+    let npmRes = shell.exec('npm start --prefix '+ ORACLES_COMBINE_DIR_PATH + ' ' + CROWDSALE_FILE_PATH);
+    if (npmRes.code !== 0) {
+       log.error('oracle compbine npm install failed');
+       throw new Error("failed to merge contract");
     } else {
-        shell.mv(ORACLES_COMBINE_OUT_DIR_PATH + '/CrowdSale_flat.sol',  path.join(MERGE_DIR_PATH, '/', CROWDSALE_FILE_NAME));
+        let mvRes = shell.mv(ORACLES_COMBINE_OUT_DIR_PATH + '/CrowdSale_flat.sol',  path.join(MERGE_DIR_PATH, '/', CROWDSALE_FILE_NAME));
+        if (mvRes.code !==0) {
```
Comment (edwardpku) [inline utils/build.js]: add space after !==

File: utils/build.js [hunk 2]
```diff
@@ -90,15 +90,19 @@ exports.mergeCrowdsaleContract = async function (tokenId) {
...
+            log.error('failed to move merged file');
+            throw new Error("failed to merge contract");
```
Comment (edwardpku) [inline utils/build.js]: failed to move merged file here?

**Review by edwardpku — APPROVED** [review]
looks good, ship it.

File: routes/routes.js [hunk 3]
```diff
@@ -53,25 +53,29 @@ var appRouter = function (app) {
...
+            let contractAddress = await buildUtil.deployCrowdsaleContract(tokenId, crowdsaleJSON);
 
-        log.info('deployed address:', contractAddress);
-        res.send(contractAddress);
+            log.info('deployed address:', contractAddress);
+            res.send(contractAddress);
+        } catch (e) {
+            res.status(500).send('deployment failed. ' + e.message);
+        }
     });
```
Comment (judezhu) [inline routes/routes.js:79]: yes! we need try & catch!

File: utils/build.js (hunk 1 above)
Comment (tyrelltle) [reply utils/build.js]: @edwardpku updated

File: utils/build.js (hunk 2 above)
Comment (tyrelltle) [reply utils/build.js]: ya just in case some unexpected situation like the folder permission issue

**Pull request merged**